Source/cmNinjaTargetGenerator.cxx

```diff
@@ -340,12 +340,42 @@ ::WriteCompileRule(const std::string& lang)
   vars.ObjectFileDir = "$OBJECT_FILE_DIR";
 
   cmMakefile* mf = this->GetMakefile();
+  
+  const char* forceRspFile = "CMAKE_NINJA_FORCE_RESPONSE_FILE";
+  bool useResponseFile = mf->IsDefinitionSet(forceRspFile) ||
+                         cmSystemTools::GetEnv(forceRspFile) != 0;
```

Comment:
Please refactor this lookup out of `WriteLinkStatement` into a dedicated helper method.  We should not need to duplicate the name of the variable or the lookup calls in every call site that needs to know.